### ISSUE TITLE
False positive result returned from directories.php

### ISSUE THREAD

**Issue by estin92** [WordPress/theme-check#193]
I've been frustratingly trying to diagnose the error "Please remove any extraneous directories like .git or .svn from the ZIP file before uploading it."

When digging into the source code, the $name variable contains the full path to the file and is ran through strpos to detect the presence of .git, .svn, .hg and .bzr. This throws a false positive though because our domain is www.hg...

`foreach ( $css_files as $name => $file ) {
			checkcount();
			if ( strpos( $name, '.git' ) !== false || strpos( $name, '.svn' ) !== false || strpos( $name, '.hg' ) !== false || strpos( $name, '.bzr' ) !== false ) $found = true;
		}`

I confirmed this is the case by removing the strpos function for .hg and the theme validated successfully. It would be good if the `$name` string could be split to only check inside the theme folders.

**Comment by GaryJones** [comment]
The use of `basename( $file )` could be used to check if the final part matches the VCS folder names.

A very quick refactoring effort - would want to see some unit tests in place, but it's a start:

```php
class DirectoriesCheck implements themecheck {
	protected $error = array();

	public function check( $php_files, $css_files, $other_files ) {
		if (
			$this->files_contain( $php_files, array( $this, 'is_git_svn' ) ) ||
			$this->files_contain( $css_files, array( $this, 'is_git_svn_hg_bzr' ) ) ||
			$this->files_contain( $other_files, array( $this, 'is_git_svn_hg_bzr' ) )
		) {
			$this->error[] = sprintf('<span class="tc-lead tc-required">' . __( 'REQUIRED', 'theme-check' ) . '</span>: ' . __( 'Please remove any extraneous directories like .git or .svn from the ZIP file before uploading it.', 'theme-check') );

			return false;
		}

		return true;
	}

	public function getError() {
		return $this->error;
	}

	protected function is_git_svn( $filepath ) {
		$vcs_folders = array( '.git', '.svn' );
		
		return in_array( $vcs_folders, basename( $filepath ), true );
	}

	protected function is_git_svn_hg_bzr( $filepath ) {
		$vcs_folders = array( '.git', '.svn', '.hg', '.bzr' );
		
		return in_array( $vcs_folders, basename( $filepath ), true );
	}

	protected function files_contain( $files, $callback ) {
		foreach ( $files as $name => $filepath ) {
			checkcount();
			if ( $callback( $filepath ) ) {
				return true;
			}
		}

		return false;
	}
}
```
